### ISSUE TITLE
Battle screech is sometimes uncastable from hand

### ISSUE THREAD

**Issue by sprangg** [magefree/mage#8462]
[[Battle Screech]] suffers from an odd issue where it's sometimes uncastable from hand. Most of the time it works ok, but this issue has now happened twice in our drafts this month.

The first time this happened, the Battle Screech was uncastable on turn 4 and the uncastability continued throughout the whole game. This was game 2 of the match, and Battle Screech was castable both in game 1 and game 3 of the match - only uncastable in game 2.

The second time there was not just one but two Battle Screeches in hand, both uncastable. However the game ended the next turn.

First time the bug happened (note how the "castable" highlight is missing from the Battle Screech):
![screech1](https://user-images.githubusercontent.com/13901691/142084788-33c33f33-67d8-4bee-a167-1312b58b87de.png)

Second time:
![screech2](https://user-images.githubusercontent.com/13901691/142084831-65922abc-863c-4567-a39e-854be4cf6cec.png)

**Comment by awjackson** [comment]
I've found a procedure to reliably reproduce this using a test deck consisting of 18 Plains and 22 Battle Screech, against an AI goldfish opponent.

1. Play lands until you have four Plains on the battlefield, enough to cast a Battle Screech from your hand.
2. Cast two Battle Screeches from your hand so that you have 4 creatures in play, enough to cast Battle Screech for its flashback cost.
3. Untap.
4. Cast a Battle Screech from your graveyard while you have one or more Battle Screech cards in hand and while your lands are untapped. After this flashbacked Battle Screech resolves, all Battle Screech cards in your hand will become uncastable (not highlighted, not clickable) and will remain so for the remainder of the game. However, if you draw another Battle Screech after triggering the bug, *that card* will be castable.

**Comment by sprangg** [comment]
Nice test! The fact that you can reliably reproduce it that way, and that the only truly special thing about Battle Screech is the flashback cost, leads me to believe that the castability criterion "have three untapped white creatures" is sometimes checked while the card is in hand, and if the result of the check is negative the card will stay uncastable for the rest of the game.

**Comment by awjackson** [comment]
In the current beta, the sequence of actions described above no longer reproduces the bug. Additionally, the XDHS has been drafting Vintage Masters on beta for the past week (a set where Battle Screech is a common) and nobody encountered issues with the card. I think this might have somehow gotten fixed between September and now, though I have no idea what change might have fixed it. Perhaps someone with a lot of spare time can do a git bisection?

**Comment by awjackson** [comment]
[Test mode](https://github.com/magefree/mage/wiki/Development-Testing-Tools) config for bisection test:
```
[init]
battlefield:Human:Plains:8
battlefield:Human:Glory Seeker:3
hand:Human:Battle Screech:2
```
Press the smiley button to add the listed cards to your battlefield and hand. Cast one of the two Battle Screeches that were added to your hand. At this point, you have four untapped plains, a Battle Screech in hand and a Battle Screech in the graveyard. In 1.4.50 from last September, the remaining Battle Screech in hand is *not castable* (bug present). In current 1.4.51beta, the Battle Screech in hand *is* castable (bug apparently fixed). Objective: find the commit that caused the behaviour to change.

**Comment by Alex-Vasile** [comment]
> Git bisect indicates that this was fixed by [de3f7cb](https://github.com/magefree/mage/commit/de3f7cbbfbd386db30c02fda22ed730ba030388e). I'm not exactly sure how it's related (`FlashbackAbility` doesn't use the copy constructor of `SpellAbility`) but I'm going to close this.

The copy constructor for `FlashbackAbility` calls `super(Ability)` which is `SpellAbility`'s copy constructor, which in turn calls `ActivatedAbilityImpl`'s copy constructor, which in turn calls `AbilityImpl`'s copy constructor (which is where that commit made the changes).
